azure-mgmt-scheduler/src/main/java/com/microsoft/azure/management/scheduler/implementation/JobImpl.java

```diff
@@ -147,4 +250,289 @@ public JobImpl withState(JobState state) {
 
         return this;
     }
+
+    @Override
+    public JobImpl minutes() {
+        if (this.inner().properties() == null) {
```

Comment:
considering that these 3 or so lines of code are repeated several times across the next few withers, wouldn't it be cleaner to just have a private `ensureProperties()` function which returne the inner properties object, ensure one is present?
And then simply refer to it in the specifici wither impls as:
`ensureProperties().withRecurrence(...)`
etc.